govcd/region_test.go

```diff
@@ -0,0 +1,161 @@
+//go:build tm || functional || ALL
+
+/*
+ * Copyright 2024 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v3/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_TmRegion(check *C) {
+	skipNonTm(vcd, check)
+	sysadminOnly(vcd, check)
+
+	vc, vcCreated, nsxtManager, nsxtManagerCreated := getOrCreateVcAndNsxtManager(vcd, check)
+	supervisor, err := vc.GetSupervisorByName(vcd.config.Tm.VcenterSupervisor)
+	check.Assert(err, IsNil)
+
+	r := &types.Region{
+		Name: check.TestName(),
+		NsxManager: &types.OpenApiReference{
+			ID: nsxtManager.NsxtManagerOpenApi.ID,
+		},
+		Supervisors: []types.OpenApiReference{
+			{
+				ID:   supervisor.Supervisor.SupervisorID,
+				Name: supervisor.Supervisor.Name,
+			},
+		},
+		StoragePolicies: []string{vcd.config.Tm.VcenterStorageProfile},
+		IsEnabled:       true,
+	}
+
+	createdRegion, err := vcd.client.CreateRegion(r)
+	check.Assert(err, IsNil)
+	check.Assert(createdRegion.Region, NotNil)
+	AddToCleanupListOpenApi(createdRegion.Region.ID, check.TestName(), types.OpenApiPathVcf+types.OpenApiEndpointRegions+createdRegion.Region.ID)
+
+	check.Assert(createdRegion.Region.Status, Equals, "READY") // Region is operational
+
+	// Get By Name
+	byName, err := vcd.client.GetRegionByName(r.Name)
+	check.Assert(err, IsNil)
+	check.Assert(byName, NotNil)
+
+	// Get By ID
+	byId, err := vcd.client.GetRegionById(createdRegion.Region.ID)
+	check.Assert(err, IsNil)
+	check.Assert(byId, NotNil)
+
+	check.Assert(byName.Region, DeepEquals, byId.Region)
+
+	// Get All
+	allRegions, err := vcd.client.GetAllRegions(nil)
+	check.Assert(err, IsNil)
+	check.Assert(allRegions, NotNil)
+	check.Assert(len(allRegions) > 0, Equals, true)
+
+	// TODO: TM: No Update so far
+	// Update
+	// createdRegion.Region.IsEnabled = false
+	// updated, err := createdRegion.Update(createdRegion.Region)
+	// check.Assert(err, IsNil)
+	// check.Assert(updated, NotNil)
+
+	// Delete
+	err = createdRegion.Delete()
+	check.Assert(err, IsNil)
+
+	notFoundByName, err := vcd.client.GetRegionByName(createdRegion.Region.Name)
+	check.Assert(ContainsNotFound(err), Equals, true)
+	check.Assert(notFoundByName, IsNil)
+
+	// Cleanup
+	if vcCreated {
+		err = vc.Disable()
+		check.Assert(err, IsNil)
+		err = vc.Delete()
+		check.Assert(err, IsNil)
+	}
+
+	if nsxtManagerCreated {
+		err = nsxtManager.Delete()
+		check.Assert(err, IsNil)
+	}
+}
+
+// getOrCreateVcAndNsxtManager will check configuration file
```

Comment:
Nice and useful!
```suggestion
// getOrCreateVcAndNsxtManager will check configuration file and create vCenter and NSX-T Manager if stated in the 'createVcenter' and 'createNsxtManager' properties and they are not present in TM. Otherwise it just retrieves them
```

Reply:
Changed the doc. I am improving it a bit further with VDC work, but this has to somehow become a core component that is convenient to use.